fsttest/recorder.go

```diff
@@ -28,12 +29,18 @@ func NewRecorder() *ResponseRecorder {
 
 // Header returns the response headers to mutate within a handler.
 func (r *ResponseRecorder) Header() fsthttp.Header {
-	return r.HeaderMap
+	if !r.headersDone {
+		return r.HeaderMap
+	}
+	return r.HeaderMap.Clone()
```

Comment:
Do you mind adding a comment here to explain the purpose?  I'm not sure it's obvious without additional context.

Reply:
This is a good point and I agree. Being part of our test harness, I was initally being more liberal about the lack of comment.

Reply:
Sure, I will update the PR.

Reply:
Done. Let me know if that works for you.